pkg/framework/guest.go

```diff
@@ -25,42 +24,44 @@ const (
 )
 
 type GuestConfig struct {
-	ClusterName  string
+	Logger micrologger.Logger
+
+	ClusterID    string
 	CommonDomain string
-	Logger       micrologger.Logger
 }
 
 type Guest struct {
-	clusterName  string
-	commonDomain string
-	logger       micrologger.Logger
+	logger micrologger.Logger
 
 	g8sClient  versioned.Interface
 	k8sClient  kubernetes.Interface
 	restConfig *rest.Config
+
+	clusterID    string
+	commonDomain string
 }
 
 func NewGuest(config GuestConfig) (*Guest, error) {
-	if config.ClusterName == "" {
-		// If config.ClusterName is not defined, default to environment variable.
-		config.ClusterName = os.Getenv("CLUSTER_NAME")
```

Comment:
Env magic should only ever be done on program initialization. Not somewhere in the middle. Having it configurable is way easier to maintain. Also the compiler can do its part and help solving problems with the code. Also the responsibility of who has to wire it is way more clear.